editor/src/core/model/element-metadata-utils.ts

```diff
@@ -1839,6 +1838,57 @@ function fillSpyOnlyMetadataWithFramesFromChildren(
     }
   })
 
+  const elementsWithoutParentData = Object.keys(fromSpy).filter((p) => {
+    const parentLayoutSystem = fromDOM[p]?.specialSizeMeasurements.parentLayoutSystem
+    return parentLayoutSystem == null
+  })
+
+  fastForEach(elementsWithoutParentData, (pathStr) => {
+    const spyElem = fromSpy[pathStr]
+    const sameThingFromWorkingElems = workingElements[pathStr]
+    const children = findChildrenInDomRecursively(pathStr)
+    if (children.length === 0) {
+      return
+    }
+
+    const childrenFromWorking = children.map((child) => {
+      const childPathStr = EP.toString(child.elementPath)
+      const fromWorkingElements = workingElements[childPathStr]
+      if (fromWorkingElements == null) {
+        return child
+      } else {
+        return fromWorkingElements
+      }
+    })
+
+    const parentLayoutSystemFromChildren = childrenFromWorking.map(
+      (c) => c.specialSizeMeasurements.parentLayoutSystem,
+    )
+    const parentFlexDirectionFromChildren = childrenFromWorking.map(
+      (c) => c.specialSizeMeasurements.parentFlexDirection,
+    )
+    const immediateParentBoundsFromChildren = childrenFromWorking.map(
+      (c) => c.specialSizeMeasurements.immediateParentBounds,
+    )
+
+    workingElements[pathStr] = {
+      ...spyElem,
+      ...sameThingFromWorkingElems,
+      specialSizeMeasurements: {
+        ...spyElem.specialSizeMeasurements,
+        parentLayoutSystem: allElemsEqual(parentLayoutSystemFromChildren)
```

Comment:
would it be nice to have a helper function for "if all elements equal, use their value, otherwise fall back to..."? or is this readable and good enough?

Reply:
(resolving, we agreed to leave it as is, since the code is still simple enough to read)